quinn-proto/src/connection.rs

```diff
@@ -2488,6 +2404,17 @@ impl From<ConnectionError> for io::Error {
     }
 }
 
+impl From<state::CloseReason> for ConnectionError {
```

Comment:
Where is this used?

Reply:
Using code you suggested in a previous comment in `Connection::handle_connected_inner -> Result<State, ConnectionError>`:

`let closed = self.process_payload(ctx, now, number, payload.into(), &mut state.tls)?;`

`Connection::process_payload` returns a `Result<bool, state::CloseReason>`, whereas `handle_connected_inner` returns a `Result<State, ConnectionError>`, so we need a way to get convert that `CloseReason` into a `ConnectionError`.

Will this be fine, or is there a better solution that you would suggest?

Reply:
That makes sense to me. I was mainly curious where it was used in your original diff.